mocha.el

```diff
@@ -54,21 +53,23 @@ If MOCHA-FILE is specified run just that file otherwise run MOCHA-PROJECT-TEST-D
           mocha-options " "
           (or mocha-file mocha-project-test-directory)))
 
-;;;###autoload
 (defun mocha-run (&optional mocha-file)
   "Run mocha in a compilation buffer.
 
 If MOCHA-FILE is specified run just that file otherwise run MOCHA-PROJECT-TEST-DIRECTORY"
   (compile (mocha-command mocha-file))
-  (kill-buffer "*mocha tests*")
+  (when (get-buffer "*mocha tests*")
+    (kill-buffer "*mocha tests*"))
   (pop-to-buffer "*compilation*")
   (rename-buffer "*mocha tests*"))
 
+;;;###autoload
```

Comment:
> So is the convention to just put autoloads on any public interactive functions?

autoload cookie should be put on functions/commands which are used for the first time of your package. it should not be put on the functions which is called after other functions of your package called.

For example a major-mode package, autoload cookie should be put on major-mode function(like `python-mode`), while it should not be put on its commands(like `python-indent-shift-left`), because command functions(like `python-indent-shift-left`)  will be called after major-mode function(like `python-mode`) called.